### PR TITLE
[NPU] [Diffusion] Performance Optimization for LTX-2 Model

### PR DESCRIPTION
**Pull request by gjsheu**
## Motivation

This PR reduces E2E latency by approximately 27% on NPU and 3% on GPU.
Before merging, ensure that PR https://github.com/sgl-project/sgl-kernel-npu/pull/431 has been successfully merged.

## Modifications

1. Replace torch.nn.RMSNorm with custom RMSNorm.
2. Integrate fused_rmsnorm_without_weight from sgl_kernel_npu.
3. Set the mask to None when it consists entirely of ones to prevent FlashAttention (FA) from degrading into smaller, less efficient operators.

## Test Script
``` shell
# NPU
sglang generate --model-path /home/weights/LTX-2/ --prompt "A beautiful sunset over the ocean" --sp-degree 2 --num-gpus 2 --warmup true
# GPU
sglang generate --model-path /home/weight/LTX-2/ --prompt "A beautiful sunset over the ocean" --num-gpus 1 --warmup true
```

## ACC Tests
before:

after:

## Speed Tests and Profiling
NPU：
E2E before the pr: 102.74s
E2E after the pr: 75.69s
GPU：
E2E before the pr: 136.62s
E2E after the pr: 131.70s

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
4. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
5. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
6. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #27686993200](https://github.com/sgl-project/sglang/actions/runs/27686993200)
Latest PR Test (Extra): :x: [Run #27686992517](https://github.com/sgl-project/sglang/actions/runs/27686992517)

### CODE REVIEW

**Comment by ping1jing2** [comment]
please resovle lint issue first and let me think do we have a way to remove _is_npu from the code

File: python/sglang/multimodal_gen/runtime/models/dits/ltx_2.py [hunk 1]
```diff
@@ -392,7 +398,14 @@ def forward(
...
 def rms_norm(x: torch.Tensor, eps: float) -> torch.Tensor:
-    return F.rms_norm(x, normalized_shape=(x.shape[-1],), eps=eps)
+    if _is_npu:
+        from sgl_kernel_npu.norm.rmsnorm_without_weight import (
+            fused_rmsnorm_without_weight,
+        )
+
+        return fused_rmsnorm_without_weight(x, eps)
+    else:
+        return F.rms_norm(x, normalized_shape=(x.shape[-1],), eps=eps)
```
Comment (Makcum888e) [inline python/sglang/multimodal_gen/runtime/models/dits/ltx_2.py]: could you create a new custom_op for avoid is_npu here?

Comment (Makcum888e) [reply python/sglang/multimodal_gen/runtime/models/dits/ltx_2.py]: in python/sglang/multimodal_gen/runtime/layers/layernorm.py
add new `class RMSNormNoWeigth(CustomOp):` and use it here

Comment (gjsheu) [reply python/sglang/multimodal_gen/runtime/models/dits/ltx_2.py]: Done. I've implemented RMSNormNoWeight(CustomOp) as suggested and replaced the condition here.

**Comment by gjsheu** [comment]
> please resovle lint issue first and let me think do we have a way to remove _is_npu from the code

I have removed _is_npu based on the feedback.

**Review by Makcum888e — APPROVED** [review]
LGTM

**Review by Makcum888e — APPROVED** [review]
LGTM

**Review by ping1jing2 — APPROVED** [review]
